packages/amplify-e2e-tests/src/__tests__/api_2.test.ts

```diff
@@ -179,14 +181,17 @@ describe('amplify add api (GraphQL)', () => {
     await addApiWithSchemaAndConflictDetection(projRoot, 'simple_model.graphql');
     await amplifyPush(projRoot);
 
+    meta = getProjectMeta(projRoot);
+
     const { output } = meta.api[name];
     const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
-    const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, meta.providers.awscloudformation.Region);
 
     expect(GraphQLAPIIdOutput).toBeDefined();
     expect(GraphQLAPIEndpointOutput).toBeDefined();
     expect(GraphQLAPIKeyOutput).toBeDefined();
 
+    const { graphqlApi } = await getAppSyncApi(GraphQLAPIIdOutput, meta.providers.awscloudformation.Region);
+
     expect(graphqlApi).toBeDefined();
     expect(graphqlApi.apiId).toEqual(GraphQLAPIIdOutput);
```

Comment:
Why?

Reply:
Did not change the test logic just fixed the order of operations.